tests/unittests/PartitionerTest.cpp

```diff
@@ -385,3 +388,78 @@ TEST_F(PartitionerTest, SelectRepFunc) {
   auto err = myPartitioner.Partition();
   EXPECT_FALSE(errToBool(std::move(err)));
 }
+
+template <BackendKind kind, glow::Kinded::Kind unsupportedOpKind>
+class MockBackend : public Backend {
+  class MockFunction : public CompiledFunction {
+  public:
+    MockFunction(const runtime::RuntimeBundle &bundle)
+        : CompiledFunction(bundle) {}
+
+    void execute(ExecutionContext *) override {}
+
+    BackendKind getCompileBackendKind() const override {
+      return BackendKind::Interpreter;
+    }
+  };
+
+  BackendKind getBackendKind() const override { return kind; }
+
+  std::string getBackendName() const override { return "MockBackend"; }
+
+  std::unique_ptr<CompiledFunction>
+  compile(Function *F, const BackendOptions &opts) const override {
+    return llvm::make_unique<MockFunction>(runtime::RuntimeBundle::create(*F));
+  }
+
+  bool isOpSupported(const NodeInfo &NI) const override {
+    if (NI.getKind() == unsupportedOpKind) {
+      return false;
+    }
+    return true;
+  }
+
+  bool shouldLower(const Node *N) const override { return false; }
+
+  bool generateInst(Node *N, IRGenVisitor &irgen) const override {
+    return false;
+  }
+};
+
```

Comment:
Add a comment explaining why these classes are needed and what they try to model.